distributed/scheduler.py

```diff
@@ -2315,19 +2321,49 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         """
         Decide on a worker for task *ts*.  Return a WorkerState.
         """
+        if not self._workers_dv:
+            return None
+
         ws: WorkerState = None
+        group: TaskGroup = ts._group
         valid_workers: set = self.valid_workers(ts)
 
         if (
             valid_workers is not None
             and not valid_workers
             and not ts._loose_restrictions
-            and self._workers_dv
         ):
             self._unrunnable.add(ts)
             ts.state = "no-worker"
             return ws
 
+        # Group is larger than cluster with few dependencies? Minimize future data transfers.
+        if (
+            valid_workers is None
+            and len(group) > self._total_nthreads * 2
+            and sum(map(len, group._dependencies)) < 5
+        ):
+            ws: WorkerState = group._last_worker
+            tasks_per_thread = len(group) / self._total_nthreads
+
+            if not (
+                ws
+                and ws._address in self._workers_dv
+                and ws._occupancy / ws._nthreads / self.get_task_duration(ts)
```

Comment:
Wait, don't we want to compare this to something else?  This only works if there is no other occupancy around.  

In my occupancy PR I compared this worker to the worker that would have been chosen if we didn't go this route.

I think that this would fail your `x - x.mean()` case.

Reply:
If you can revert to your counting approach quickly then no objection from me.

Reply:
Correct, this doesn't work if there's any initial occupancy. My concern with the comparison approach was that I wasn't sure how representative some worker A's occupancy was of worker B's initial occupancy, when we're only picking worker A from those that hold dependencies (assuming #4925 is a ways out).

Interestingly the `x - x.mean()` case still works fine. The problem arises when I add `fs = client.map(time.sleep, [0.2] * 600, pure=False)` before computing `x - x.mean()`. Then we start round-robining too quickly and split neighbors across workers.

I can go back to the counting approach. I had one other idea: we could store `TaskGroup._initial_worker_occupancy` when we pick the new worker, and subtract that out in the calculation. Then, we really do end up just using occupancy as a count. But I'm not sure there's any advantage to that over the counting; it's probably just a more roundabout way of doing the same thing.

Reply:
> TaskGroup._initial_worker_occupancy

This sounds like a more obfuscated count to me.  Let's just count.